### PR TITLE
Automated Deployments

### LINKED ISSUE #16: Automated deployments
For every push to `main`, we should:
1. Build a new image and push it to ACR
2. Deploy the latest image to ACI

### PR DESCRIPTION
**Pull request by amondal2**
Closes #16 --

This PR required a bit of testing (hence the commit history), but it is working as expected now. The general workflow:

1. We configured a `production` environment in the Github repo that holds the Azure secrets
2. When a push event is triggered to the master branch:
- The GH runner pulls the secrets
- Logs into Azure CLI
- Builds the latest image, tagged with the git commit SHA 
- Pushes the updated image to ACR
- Restarts the container application with the updated SHA; this creates a new revision and disables the old one

We should now be able to replicate this workflow for any Dockerized application moving forward.

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - name: Configure Docker
+      uses: docker/setup-buildx-action@v3
+
+    - name: Azure ACR login
+      uses: azure/docker-login@v2
+      with:
+        login-server: cfaprdconfigvalidation.azurecr.io
```
Comment (natemcintosh) [inline .github/workflows/deploy.yaml:24]: what is a `login-server`?

**Review by natemcintosh — APPROVED** [review]
Wow, love that this CI file is so small, yet accomplishes so much

**Review by zsusswein — APPROVED** [review]
Do you want to do some kind of cleanup of the old tagged images?

This is super cool. Love that it's so lightweight.

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (amondal2) [reply .github/workflows/deploy.yaml:24]: This is just the URL for the container registry where we are pushing images. You can see in the Azure portal under "Container Registries"

**Comment by amondal2** [comment]
@zsusswein -- Yes, let me think about how to handle that separately.

**Comment by zsusswein** [comment]
Should this link to #16?

**Comment by amondal2** [comment]
Yep, whoops let me update

**Pull request merged**